### PR TITLE
Update: Clone widget

### PR DESCRIPTION
**Pull request by daniellevinson**
## Description

Clone widget

## Proposed Changes

- clone widget
- scroll to bottom when a new widget is added via add or clone

## Screenshots

https://user-images.githubusercontent.com/13946669/170116259-177cfb53-dab6-430c-9d85-f9a0ace75e32.mov

## License

I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: packages/dashboards/tests/acceptance/clone-widget-test.js [hunk 1]
```diff
@@ -0,0 +1,246 @@
...
+  test('visiting the clone widget route with no edit permissions', async function (assert) {
+    await visit('/dashboards/4');
+
+    assert.dom('.navi-widget').exists({ count: 1 }, 'dashboard initially contains only 1 widget');
+
+    assert.dom('.navi-widget__clone-btn').doesNotExist('clone widget button does not exist');
+
+    await visit('/dashboards/4/widgets/6/clone');
+
+    assert.equal(currentURL(), '/dashboards/4/view', 'redirects to dashboards/4/view');
```
Comment (kevinhinterlong) [inline packages/dashboards/tests/acceptance/clone-widget-test.js:54]: Could send a notification that says they do not have permission to clone ___

**Review by kevinhinterlong — COMMENTED** [review]
Some edge cases that aren't handled
- invalid lastWidgetAdded id is not removed
- cloning a widget and deleting it does not clear the query param

**Comment by daniellevinson** [comment]
> Some edge cases that aren't handled
> 
> * invalid lastWidgetAdded id is not removed
> * cloning a widget and deleting it does not clear the query param

yeah I didn't handle cases like these since they shouldn't break anything or change the UI behavior
but I don't mind adding some checks

File: packages/dashboards/tests/acceptance/dashboards-test.js [hunk 2]
```diff
@@ -195,6 +195,90 @@ module('Acceptance | Dashboards', function (hooks) {
...
+      `/dashboards/1/view?filters=${filters}`,
+      `lastAddedWidgetId query param that is not in the dashboad is removed`
+    );
+
+    await click('.navi-dashboard__revert-button'); // to avoid unsaved changes dialog
+    await visit(`/dashboards/1/view?filters=${filters}&lastAddedWidgetId=100`);
+    assert.equal(
+      currentURL(),
+      `/dashboards/1/view?filters=${filters}`,
+      `zero lastAddedWidgetId query param that is not in the dashboad is removed (view route)`
```
Comment (daniellevinson) [inline packages/dashboards/tests/acceptance/dashboards-test.js:278]:
```suggestion
      `lastAddedWidgetId query param that is not in the dashboad is removed (view route)`
```
